folsom/src/test/java/com/spotify/folsom/reconnect/ReconnectingClientTest.java

```diff
@@ -203,4 +211,44 @@ public void registerForConnectionChanges(ConnectionChangeListener listener) {
       }
     }
   }
+
+  @Test
+  public void testFiledescriptorLeak() throws InterruptedException, TimeoutException {
+    final EmbeddedServer embeddedServer = new EmbeddedServer(false);
+    final MemcacheClient<String> client =
+        MemcacheClientBuilder.newStringClient()
+            .withAddress("localhost", embeddedServer.getPort())
+            .connectAscii();
+    client.awaitFullyConnected(1, TimeUnit.MINUTES);
+
+    // Warmup (maybe not needed?)
+    for (int i = 0; i < 10; i++) {
+      triggerReconnect(embeddedServer, client);
+    }
+
+    long fdCountPre = getFDCount();
+    for (int i = 0; i < 10; i++) {
+      triggerReconnect(embeddedServer, client);
+    }
+    long fdCountPost = getFDCount();
+
+    assertEquals(fdCountPost, fdCountPre);
+  }
+
+  private void triggerReconnect(
+      final EmbeddedServer embeddedServer, final MemcacheClient<String> client)
+      throws TimeoutException, InterruptedException {
+    embeddedServer.stop();
+    client.awaitFullyDisconnected(1, TimeUnit.MINUTES);
+    embeddedServer.start();
+    client.awaitFullyConnected(1, TimeUnit.MINUTES);
+  }
+
+  private long getFDCount() {
+    OperatingSystemMXBean os = ManagementFactory.getOperatingSystemMXBean();
+    if (os instanceof UnixOperatingSystemMXBean) {
+      return ((UnixOperatingSystemMXBean) os).getOpenFileDescriptorCount();
+    }
+    throw new UnsupportedOperationException("Unavailable");
```

Comment:
you could also use one of the junit [Assume](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html) methods to ignore the test if certain conditions are not met rather than throw an exception and fail the test

Reply:
Good point! I should have done that - will tweak later.